OpenGL/Constructs/VBO.cs

```diff
@@ -118,6 +128,20 @@ public VBO(T[] Data)
         {
         }
 
+        private int GetStructSize()
+        {
+            Dictionary<Type, int> structSizes = new Dictionary<Type, int>()
+            {
+                [typeof(int)] = 1,
+                [typeof(float)] = 1,
+                [typeof(Vector2)] = 2,
+                [typeof(Vector3)] = 3,
+                [typeof(Vector4)] = 4,
+            };
+
+            return structSizes[typeof(T)];
+        }
+
```

Comment:
Won't this create a new Dictionary<Type, int> every time this method is called?  I would prefer this be translated to a switch statement, or if you prefer the Dictionary approach then to make the Dictionary static and initialize it a single time.